w5500/netdev.go

```diff
@@ -0,0 +1,457 @@
+package w5500
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+	"runtime"
+	"time"
+
+	"tinygo.org/x/drivers/netdev"
+)
+
+type socket struct {
+	sockn    uint8
+	protocol uint8
+	port     uint16
+	inUse    bool
+	closed   bool
+}
+
+func (s *socket) setSockn(n int) *socket {
+	s.sockn = uint8(n)
+	return s
+}
+
+func (s *socket) setProtocol(proto byte) *socket {
+	s.protocol = proto
+	return s
+}
+
+func (s *socket) setPort(port uint16) *socket {
+	s.port = port
+	return s
+}
+
+func (s *socket) setInUse(inUse bool) *socket {
+	s.inUse = inUse
+	return s
+}
+
+func (s *socket) setClosed(closed bool) *socket {
+	s.closed = closed
+	return s
+}
+
+func (s *socket) reset() {
+	s.protocol = 0
+	s.port = 0
+	s.inUse = false
+	s.closed = false
+}
+
+// GetHostByName resolves the given host name to an IP address.
+func (d *Device) GetHostByName(name string) (netip.Addr, error) {
+	d.mu.Lock()
+	dns := d.dns
+	d.mu.Unlock()
+
+	if dns == nil {
+		return netip.Addr{}, netdev.ErrNotSupported
+	}
+	return dns(name)
+}
+
+func (d *Device) Socket(domain int, stype int, protocol int) (int, error) {
+	if domain != netdev.AF_INET {
+		return -1, netdev.ErrFamilyNotSupported
+	}
+	switch {
+	case stype == netdev.SOCK_STREAM && protocol == netdev.IPPROTO_TCP:
+	case stype == netdev.SOCK_DGRAM && protocol == netdev.IPPROTO_UDP:
+	default:
+		return -1, fmt.Errorf("unsupported socket type %d and protocol %d", stype, protocol)
+	}
+
+	var proto byte
+	switch protocol {
+	case netdev.IPPROTO_TCP:
+		proto = 1 // TCP
+	case netdev.IPPROTO_UDP:
+		proto = 2 // UDP
+	default:
+		return -1, netdev.ErrNotSupported
+	}
+
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	sockfd, sock, err := d.nextSocket()
+	if err != nil {
+		return -1, err
+	}
+
+	d.openSocket(sock.sockn, proto)
+
+	sock.setProtocol(proto).setInUse(true)
+	return sockfd, nil
+}
+
+func (d *Device) openSocket(sockn uint8, proto byte) {
+	d.writeByte(sockMode, sockAddr(sockn), proto&0x0F)
+}
+
+func (d *Device) Bind(sockfd int, ip netip.AddrPort) error {
+	// The IP address is irrelevant. The configured ip will always be used.
+	port := ip.Port()
+	if port < 1 || port > 65535 {
```

Comment:
```suggestion
	if port < 1 {
```
redundant check, as we have `uint16`.
And three other similar places in this file.

Reply:
Hm, shall we have this conditional at all?